chrome/content/zotero/xpcom/utilities.js

```diff
@@ -179,9 +179,13 @@ Zotero.Utilities = {
 				var lastName = author;
 			}
 		} else {
-			var spaceIndex = author.lastIndexOf(" ");
-			var lastName = author.substring(spaceIndex+1);
-			var firstName = author.substring(0, spaceIndex);
+			// Don't parse "Firstname Lastname [Country]" as "[Country], Firstname Lastname"
+			var spaceIndex = author.length;
+			do {
+				spaceIndex = author.lastIndexOf(" ", spaceIndex-1);
+				var lastName = author.substring(spaceIndex + 1);
+				var firstName = author.substring(0, spaceIndex);
+			} while (!/\w/.test(lastName[0]) && spaceIndex > 0)
```

Comment:
1) Any changes here would need supporting tests.

2) `\w` and the like aren't really usable for user input (as surely you would agree). This would fail on any family name that started with an extended character. XRegExp and its much more extensive character classes, including one for punctuation, are available as `this.XRegExp` in `Zotero.Utilities`, though.